### DISCUSSION TITLE
Getting nvidia GPUs to run in jail

### DISCUSSION THREAD

**Discussion by MildlyInterested** [Jip-Hop/jailmaker#43] [Q&A] [answered]
While trying to install the nvidia driver specific to my GPU (GT 710)  I'm running in to this error:
```bash
nvidia-installer log file '/var/log/nvidia-installer.log'
creation time: Sat Oct 14 16:34:59 2023
installer version: 470.199.02

PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin

nvidia-installer command line:
    ./nvidia-installer

Using: nvidia-installer ncurses v6 user interface
-> Detected 12 CPUs online; setting concurrency level to 12.
-> Installing NVIDIA driver version 470.199.02.
-> Performing CC sanity check with CC="/usr/bin/cc".
-> Performing CC check.
ERROR: Unable to find the kernel source tree for the currently running kernel.  Please make sure you have installed the kernel source files for your kernel and that they are properly configured; on Red Hat Linux systems, for example, be sure you have the 'kernel-source' or 'kernel-devel' RPM installed.  If you know the correct kernel source files are installed, you may specify the kernel source path with the '--kernel-source-path' command line option.
ERROR: Installation has failed.  Please see the file '/var/log/nvidia-installer.log' for details.  You may find suggestions on fixing installation problems in the README available on the Linux driver download page at www.nvidia.com.
```

I'm assuming that kernel source tree error is due to running the install in the jail, has anyone else encountered it and if so how did you solve it?

Thanks!

**Comment by Jip-Hop** [comment, marked as answer]
You can only use GPUs which are supported by the included driver on the host OS. If your GPU doesn't work with the driver version TrueNAS SCALE ships, then you can't use it in the jail either. If your GPU is supported by SCALE (I think the upcoming 23.10 release has a newer driver version) then you can enable the GPU pass through option with jailmaker. You won't have to install anything in the jail.

**Reply by MildlyInterested** [reply]
Thanks for the fast answer! `lspci | grep NVIDIA` does report my GPU in both truenas and the jail so I was hoping I could install the drivers in the jail.
I will wait then.

**Reply by Jip-Hop** [reply]
You should not need to install drivers. What are you trying to do? Is nvidia-smi available inside the jail?

**Reply by MildlyInterested** [reply]
Not available inside the jail, all I get is `command not found`.
Outside the jail I get `NVIDIA-SMI has failed because it couldn't communicate with the NVIDIA driver. Make sure that the latest NVIDIA driver is installed and running.` Maybe should have tried that earlier.
